tools/build_script_generator/scons/resources/SConscript.in

```diff
@@ -16,6 +16,11 @@ profile = ARGUMENTS.get("profile", "release")
 env["CONFIG_PROFILE"] = profile
 env["BUILDPATH"] = join(env["CONFIG_BUILD_BASE"], profile)
 env["BASEPATH"] = abspath("..")
+%% if core.startswith("avr")
+env["COMPILERPREFIX"] = "avr-"
+%% elif core.startswith("cortex-m")
+env["COMPILERPREFIX"] = "arm-none-eabi-"
+%% endif
```

Comment:
This won't scale well for e.g. RISC-V and/or non-GCC compilers, but is ok for now...